daemon/metrics.go

```diff
@@ -40,3 +49,62 @@ func init() {
 	imageActions = ns.NewLabeledTimer("image_actions", "The number of seconds it takes to process each image action", "action")
 	metrics.Register(ns)
 }
+
+type stateCounter struct {
+	running int32
+	stopped int32
+	paused  int32
+
+	// This is used so we don't mess with counters until state is synced from
+	// containerd after daemon restart
+	ready int32
+}
+
+func (ctr *stateCounter) isReady() bool {
+	return atomic.LoadInt32(&ctr.ready) == 1
+}
+
+func (ctr *stateCounter) setReady() {
+	atomic.StoreInt32(&ctr.ready, 1)
+}
+
+func (ctr *stateCounter) inc(label string) {
+	if !ctr.isReady() {
+		return
+	}
+	switch label {
+	case "running":
```

Comment:
Mostly because the intention was to use this for prom which needs a meaningful label value where an enum doesn't cut it.